### PR TITLE
chore: Move models out of core in checkout

### PR DESCRIPTION
**Pull request by bgambocjaviniar**

### CODE REVIEW

File: feature-libs/cart/main/root/models/cart.model.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
+import { PaymentDetails } from '@spartacus/checkout/base/root';
```
Comment (znikola) [inline feature-libs/cart/main/root/models/cart.model.ts]: I'm wondering if this is a good idea - _checkout_ already depends on _cart_, but now we're making _cart_ depend on _checkout_.
Maybe this is a good sign that `PaymentDetails` actually belongs to _user_ lib?

File: feature-libs/cart/main/root/models/cart.model.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 import {
   Address,
```
Comment (znikola) [inline feature-libs/cart/main/root/models/cart.model.ts:2]: A similar question is for the _order_ lib.

File: feature-libs/cart/main/root/models/cart.model.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
   CostCenter,
   Currency,
```
Comment (znikola) [inline feature-libs/cart/main/root/models/cart.model.ts:4]: Not sure how this happened, but once we run `yarn generate:deps` and `yarn config:update` we get the checkout ts configs inside of the checkout tsconfig itself 🤔

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/cart/main/root/models/cart.model.ts (hunk 1 above)
Comment (bgambocjaviniar) [reply feature-libs/cart/main/root/models/cart.model.ts]: moving to cart

**Pull request merged**